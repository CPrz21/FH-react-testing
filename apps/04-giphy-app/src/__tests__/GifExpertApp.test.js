import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifExpertApp} from "../GifExpertApp";

describe('<GifExpertApp />', () => {
  
  test('should show component correctly', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  })
  test('should show category list', () => {
    const categories = ['developer', 'lol'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })
  
})
