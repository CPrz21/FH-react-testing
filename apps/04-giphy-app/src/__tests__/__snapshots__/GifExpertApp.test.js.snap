// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> should show category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="developer"
      key="developer"
    />
    <GifGrid
      category="lol"
      key="lol"
    />
  </ol>
</Fragment>
`;

exports[`<GifExpertApp /> should show component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
